### PR TITLE
feat: pool methods

### PR DESCRIPTION
**Pull request by uzielflorida**

### CODE REVIEW

File: src/contracts/Pool.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      try {
+        const owner = await withdrawalQueueContract.ownerOf(tokenId);
+        const available = await withdrawalQueueContract.getAvailable(tokenId);
+        if (owner !== this.wallet.account.address || !available) continue;
+        const claim = await withdrawalQueueContract.withdraw({ to: receiver, tokenId });
+        results.push({ status: FULFILLED, value: claim });
+      } catch (reason) {
+        results.push({ status: REJECTED, reason });
+      }
+    }
```
Comment (franco-gondi) [inline src/contracts/Pool.ts:88]:
```suggestion

    .map(queue => {
      const withdrawalQueueContract = new WithdrawalQueue({
        walletClient: this.wallet,
        address: queue.contractAddress,
      });
      try {
        const owner = await withdrawalQueueContract.ownerOf(tokenId);
        const available = await withdrawalQueueContract.getAvailable(tokenId);
        if (owner !== this.wallet.account.address || !available) continue;
        const claim = await withdrawalQueueContract.withdraw({ to: receiver, tokenId });
        return { status: FULFILLED, value: claim };
      } catch (reason) {
        return { status: REJECTED, reason };
      }
    }
```

File: src/contracts/Pool.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  async mint({ amount, receiver }: { amount: bigint; receiver: Address }) {
+    const txHash = await this.safeContractWrite.mint([amount, receiver]);
+    return {
+      txHash,
+      waitTxInBlock: async () => {
+        const receipt = await this.bcClient.waitForTransactionReceipt({
+          hash: txHash,
+        });
+
+        const filter = await this.contract.createEventFilter.Deposit({});
```
[the remark this answers is not in the document]
Comment (uzielflorida) [reply src/contracts/Pool.ts:46]: same event

File: src/gondi.ts [hunk 3]
```diff
@@ -1047,6 +1048,101 @@ export class Gondi {
...
+    if (isDefined(shareAmount)) {
+      return poolContract.redeem({ amount: shareAmount, ...receiverOwnerConfig });
+    }
+
+    throw new Error('Invalid pool withdraw');
+  }
+
+  async poolClaim({
+    address,
+    tokenIdsForEachQueue,
```
Comment (uzielflorida) [inline src/gondi.ts:1133]: review this parameter name

File: src/contracts/Pool.ts [hunk 4]
```diff
@@ -153,4 +154,13 @@ export class Pool extends BaseContract<typeof poolAbi> {
         }),
     );
   }
+
+  async getMaxOfferDuration() {
+    const maxQueues = await this.contract.read.getMaxTotalWithdrawalQueues();
+    const getMinTimeBetweenWithdrawalQueues =
+      await this.contract.read.getMinTimeBetweenWithdrawalQueues();
+    const maxOfferDurationSeconds =
+      maxQueues * getMinTimeBetweenWithdrawalQueues - BigInt(daysToSeconds(7));
```
Comment (mishuagopian) [inline src/contracts/Pool.ts:163]: What's this 7 value? Will it be hardcoded in the contract?

Comment (uzielflorida) [reply src/contracts/Pool.ts:163]: It's hardcoded in the contract, yep

File: package.json [hunk 5]
```diff
@@ -34,6 +34,7 @@
   "devDependencies": {
     "@graphql-codegen/add": "^5.0.0",
     "@graphql-codegen/cli": "^3.3.1",
+    "@graphql-codegen/schema-ast": "^2.6.1",
```
Comment (mishuagopian) [inline package.json:37]: Was this missing or what?

File: src/contracts/Pool.ts [hunk 6]
```diff
@@ -144,15 +144,15 @@ export class Pool extends BaseContract<typeof poolAbi> {
...
+        new Array(maxQueues + 1)
           .fill(0)
           .map(async (_, i) => await this.contract.read.getDeployedQueue([BigInt(i)])),
       )
     ).map(
       (queue) =>
         new WithdrawalQueue({
           walletClient: this.wallet,
-          address: queue.contractAddress,
+          address: queue.contractAddress.toLowerCase() as Address,
```
Comment (mishuagopian) [inline src/contracts/Pool.ts:155]: Maybe we can use getAddress from viem instead, to avoid explicit casting

File: package.json (hunk 5 above)
Comment (uzielflorida) [reply package.json:37]: Yep

**Review by mishuagopian — APPROVED** [review]

File: src/gondi.ts [hunk 7]
```diff
@@ -413,6 +413,14 @@ export class Gondi {
     });
   }
 
+  async withdrawalPositions({ limit = 20, cursor, ...rest }: model.ListWithdrawalPositionsProps) {
```
Comment (mishuagopian) [inline src/gondi.ts:416]: Let's bump the version but also add `  /** @internal */` to all new methods so that are not included in the docs

**Review by mishuagopian — APPROVED** [review]

**Pull request merged**